### PR TITLE
Change PR template

### PR DESCRIPTION
**Pull request by caesarsol**
I wanted to: 
- require some more details on every opened PR
- require screenshots of before/after
- remove the checkboxes, that were causing this wrong widget from appearing 
	image
- keep it simple so that one does not feel overwhelmed

Feel free to comment with anything, I'll implement the changes! :)

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md
Comment (aless10) [inline PULL_REQUEST_TEMPLATE.md]: 🚀

**Comment by caesarsol** [comment]
Added the "Steps to test" section, for logics changes or backend requiring manual testing.
Let me know if unclear or not useful!

File: PULL_REQUEST_TEMPLATE.md
Comment (aless10) [inline PULL_REQUEST_TEMPLATE.md:11]: Screenshots are below. I think that we should remove this from here

File: PULL_REQUEST_TEMPLATE.md
Comment (aless10) [inline PULL_REQUEST_TEMPLATE.md]: When I open a PR, I don't like to remove things from a template. And I think that this is similar to the one removed (with emoji instead of checkboxes.
Since below there is a section "Problem or request" this is redundant

**Comment by aless10** [comment]
[https://github.com/devspace/awesome-github-templates](https://github.com/devspace/awesome-github-templates)?

File: PULL_REQUEST_TEMPLATE.md
Comment (caesarsol) [reply PULL_REQUEST_TEMPLATE.md]: Good point! 
How could we express a "PR category" to be expressed though?
You are right that's redundant, maybe I just want to know if it's a CHANGE/FIX/REFACTOR/FEATURE, I could add it as a comment?

**Comment by aless10** [comment]
I suggest to add a `### Related Issue` section so that we can relate the PR to an open issue, so that there could be (hopefully) a 1to1 correspondence between issues and activities

File: PULL_REQUEST_TEMPLATE.md
Comment (caesarsol) [reply PULL_REQUEST_TEMPLATE.md:11]: This is the Request though, it could be performed in multiple ways. 
Examples:
- "Map is too slow"
- "Add chart X on page Y"

**Comment by caesarsol** [comment]
> I suggest to add a `### Related Issue` section so that we can relate the PR to an open issue, so that there could be (hopefully) a 1to1 correspondence between issues and activities

There is this feature in GH :) we could encourage to fill it in!
image

**Comment by caesarsol** [comment]
I re-read this old discussion:
https://accurat.slack.com/archives/C07UZ4R0U/p1594287189001100

New proposal: add a text at the end for reviewers, asking them to specify how it has been tested in the approval message. Something like:

> For PR Reviewers, please include in the Approval/Rejection message:
> - Add a brief explanation of how did you review the code: quick look or in-depth?
> - Did you test on the deploy or in local? quick or exhaustive manual testing?

Vote!

**Comment by aless10** [comment]
> > I suggest to add a `### Related Issue` section so that we can relate the PR to an open issue, so that there could be (hopefully) a 1to1 correspondence between issues and activities
> 
> There is this feature in GH :) we could encourage to fill it in!
> image

I know that, but the thing is that if you add a "this PR closes #whatever" it automatically adds the reference to the linked issues on the right, and I think that it's easier to implement.
Just to say that I think that this github feature is not so well implemented in my opinion.
I guess that the way to encourage to use this feature is to make it explicit (Explicit is better than implicit.)

File: PULL_REQUEST_TEMPLATE.md
Comment (caesarsol) [reply PULL_REQUEST_TEMPLATE.md]: Moved as shorter text in the Solution section

**Comment by caesarsol** [comment]
Implemented @aless10 proposals.

**Review by aless10 — APPROVED** [review]

**Review by lucafalasco — COMMENTED** [review]
Nice @caesarsol, some personal thoughts:
- `Problem or request` / `Solution` could be simplified in one paragraph in my opinion, most of the time the PR adds a new feature or fixes a bug, so the solution and the request could often coincide (think of "Add menu bar" for example). Of course one is free to add additional details, but I would say it needs to be really short, after all you can look at the code to see what was the fix/feature about
- I know it might go against the trend but I really don't see any additional value in always having `Screenshots` in the PR, if we already have the PR deploy I actually prefer to test the branch there myself. So, for me, it's enough to know what has changed as a text description. Also a lot of times there's no visual difference so it wouldn't make sense. 
Maybe we could just leave this up to the PR creator without the need to include the section in the template? 
- I'm not sure `Steps to test` makes sense here, it's something more related to the specific issue the PR is for, maybe?
- I like the `TODO` and `Related Issues` sections 👍 
- `For PR Reviewers Only` seems a bit unnecessary to me, in my opinion it's something we can decide together and eventually add in the review message, without the need to include it in the template.

In general, it looks a bit verbose to me.
My point is that I believe we should think of the template as a really basic/essential skeleton, and one can always add things on top of that when needed.

**Comment by caesarsol** [comment]
Hey Luke!
Going in detail in a bit, but let me say this:
1. You are right, it's a bit verbose. I'm thinking of ways to make it a bit shorter and easier, but with the same level of detail. One can always remove the non relevant sections, so I'm not worried about the length.
2. It's true that one can always add things, the point here is to ENCOURAGE to add the described information. If one thinks the screenshots are not needed, I prefer him to think twice rather than not think at all :)

**Review by ruggerocastagnola — APPROVED** [review]

**Pull request closed without merging**
